src/js/bootstrap-slider.js

```diff
@@ -1862,9 +1865,11 @@ const windowIsDefined = (typeof window === "object");
 
 				var callbackFnArray = this.eventToCallbackMap[evt];
 				if(callbackFnArray && callbackFnArray.length) {
-					for(var i = 0; i < callbackFnArray.length; i++) {
-						var callbackFn = callbackFnArray[i];
-						callbackFn(val);
+					for(let i = 0; i < callbackFnArray.length; i++) {
+						let callbackFn = callbackFnArray[i];
+						window.setTimeout(() => {
+							callbackFn(val);
```

Comment:
See my review comment below, the callbacks should be executed synchronously at the moment they're triggered, which is how `$.trigger()` behaves.

Reply:
thanks for pointing this out!

I reflected on this a bit more and I agree with you. Also, when a user triggers a mouse or keyboard based sliding event, this automatically triggers the necessary callbacks in an async fashion via the browsers native event handling mechanism.

If a user directly changes the value via the `setValue()` call, then the event handlers should execute in a sync fashion (as you indicated).

I look at another popular slider library called [noUiSlider](https://github.com/leongersen/noUiSlider), and this is how they do it as well.